### DISCUSSION TITLE
When to declare the testing resources

### DISCUSSION THREAD

**Discussion by chrislorro** [puppetlabs/opv#11] [General]
I am curious where in the implementation these type of test resources are configured

when they should be classified in a manifest / or implemented in a profile specifically for running the tests based on the resources being managed / or application being deployed

what happens when a test fails, should the code then configured a parameter, for example:

could it be coded in a way similar to the `ensure => running` attribute in the service resource

**Comment by DavidS** [comment]
> I am curious where in the implementation these type of test resources are configured
> 
> when they should be classified in a manifest / or implemented in a profile specifically for running the tests based on the resources being managed / or application being deployed

This really depends on where the "know-how" of what should be checked comes from, e.g. `apache::vhost` can have a default check already in the define, but also your profile can have one based on the hiera data that goes into the node. Those might check the same thing, or different things, depending on your specific use-case and implementation.

Where would you put it?

> what happens when a test fails, should the code then configured a parameter, for example:
> 
> could it be coded in a way similar to the `ensure => running` attribute in the service resource

Since there is no clear way on how to reach the "desired state" expressed by a check resource, I'm not sure that `ensure` is the correct word here. The [current design](https://github.com/puppetlabs/opv/issues/1) for `check_http` for example has `expected_statuses` instead, expressing what the author of the code expected.

In what ways would you expect tests like this to fail?
